x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskCleaner.java

```diff
@@ -48,9 +61,17 @@ public void clusterChanged(final ClusterChangedEvent event) {
         if (event.localNodeMaster() == false) {
             return;
         }
+        final Metadata metadata = event.state().metadata();
+        final PersistentTasksCustomMetadata persistentTasksMetadata = metadata.custom(PersistentTasksCustomMetadata.TYPE);
+        final Metadata previousMetadata = event.previousState().metadata();
+        if (metadata.indices() == event.previousState().getMetadata().indices()
+            && persistentTasksMetadata == previousMetadata.custom(PersistentTasksCustomMetadata.TYPE)
+            && event.previousState().nodes().isLocalNodeElectedMaster()
+            && event.blocksChanged() == false) {
```

Comment:
NIT: this condition packs a lot of checks together. Consider moving `event.blocksChanged() == false` to the first one in this method and `event.previousState().nodes().isLocalNodeElectedMaster()` to the second one. This way conditions are spread and grouped by blocks, master and indices/tasks

Reply:
But it's an and condition? Not sure how I can move that kind of check to another condition without changing the logic?

Reply:
Yeah, disregard that